iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/metric/PipeTsFileToTabletMetrics.java

```diff
@@ -0,0 +1,256 @@
+            Tag.NAME.toString(),
+            pipeID.getPipeName(),
+            Tag.CREATION_TIME.toString(),
+            pipeID.getCreationTime(),
+            Tag.FROM.toString(),
+            pipeID.getCallerName()));
+    tabletCountMap.put(
+        pipeID,
+        metricService.getOrCreateRate(
+            Metric.PIPE_TSFILETOTABLET_TABLET_COUNT.toString(),
+            MetricLevel.IMPORTANT,
+            Tag.NAME.toString(),
+            pipeID.getPipeName(),
+            Tag.CREATION_TIME.toString(),
+            pipeID.getCreationTime(),
+            Tag.FROM.toString(),
+            pipeID.getCallerName()));
+  }
+
+  @Override
+  public void unbindFrom(final AbstractMetricService metricService) {
+    final ImmutableSet<String> pipeFullNames = ImmutableSet.copyOf(pipeIDMap.keySet());
+    for (final String pipeFullName : pipeFullNames) {
+      deregister(pipeFullName);
+    }
+    if (!pipeIDMap.isEmpty()) {
+      LOGGER.warn("Failed to unbind from pipeTsFileToTablet metrics,  pipeIDMap not empty");
+    }
+  }
+
+  private void removeMetrics(final PipeID pipeID) {
+    removeRate(pipeID);
+  }
+
+  private void removeRate(final PipeID pipeID) {
+    tsFileSizeMap.remove(pipeID);
+    metricService.remove(
+        MetricType.RATE,
+        Metric.PIPE_TSFILETOTABLET_TSFILE_SIZE.toString(),
+        Tag.NAME.toString(),
+        pipeID.getPipeName(),
+        Tag.CREATION_TIME.toString(),
+        pipeID.getCreationTime(),
+        Tag.FROM.toString(),
+        pipeID.getCallerName());
+    tabletCountMap.remove(pipeID);
+    metricService.remove(
+        MetricType.RATE,
+        Metric.PIPE_TSFILETOTABLET_TABLET_COUNT.toString(),
+        Tag.NAME.toString(),
+        pipeID.getPipeName(),
+        Tag.CREATION_TIME.toString(),
+        pipeID.getCreationTime(),
+        Tag.FROM.toString(),
+        pipeID.getCallerName());
+  }
+
+  //////////////////////////// register & deregister (pipe integration) ////////////////////////////
+
+  public void register(@NonNull final PipeID pipeID) {
+    pipeIDMap.putIfAbsent(
+        pipeID.getPipeFullName(), Collections.newSetFromMap(new ConcurrentHashMap<>()));
+    if (pipeIDMap.get(pipeID.getPipeFullName()).add(pipeID) && Objects.nonNull(metricService)) {
+      createMetrics(pipeID);
+    }
+  }
+
+  public void deregister(final String pipeFullName) {
+    if (!pipeIDMap.containsKey(pipeFullName)) {
+      return;
+    }
+    if (Objects.nonNull(metricService)) {
+      for (PipeID pipeID : pipeIDMap.get(pipeFullName)) {
+        removeMetrics(pipeID);
+      }
+    }
+    pipeIDMap.remove(pipeFullName);
+  }
+
+  public void markTsFileSize(final PipeID pipeID, final long size) {
+    if (Objects.isNull(metricService)) {
+      return;
+    }
+    final Rate rate = tsFileSizeMap.get(pipeID);
+    if (rate == null) {
+      LOGGER.info("Failed to mark pipe tsfile size, PipeID({}) does not exist", pipeID);
+      return;
+    }
+    rate.mark(size);
```

Comment:
Is it OK to use "rate"? If we use "rate" we may concentrate on "TsFileSize / s" and "TabletCount / s", rather than "per tsFile"... Seemingly we only need to record the last value? If we want to know the average of the values, or any other staticstics, the Grafana may help us to do this.